client/src/components/admin/SanitizeAllow.vue

```diff
@@ -0,0 +1,187 @@
+                                    </tr>
+                                </template>
+                            </template>
+                        </base-grid>
+                    </b-tab>
+                    <b-tab title="HTML Rendered">
+                        <base-grid :is-loaded="isLoaded" :columns="columns" id="sanitize-allow-grid">
+                            <template v-slot:rows>
+                                <template v-for="(row, localAllowedIdx) in localAllowed">
+                                    <tr :key="localAllowedIdx">
+                                        <td>{{ row.tool_name }}</td>
+                                        <td>{{ row.tool_id[0] }}</td>
+                                        <td>
+                                            <button @click="sanitizeHTML(row.ids.allowed)">Sanitize HTML</button>
+                                        </td>
+                                    </tr>
+                                </template>
+                            </template>
+                        </base-grid>
+                    </b-tab>
+                </b-tabs>
+            </b-tab>
+        </b-tabs>
+    </div>
+</template>
+
+<script>
+import BaseGrid from "./BaseGrid.vue";
+import axios from "axios";
+import { getAppRoot } from "onload/loadConfig";
+import Message from "../Message.vue";
+
+export default {
+    data() {
+        return {
+            isLoaded: false,
+            localAllowed: [],
+            localBlocked: [],
+            toolshedAllowed: [],
+            toolshedBlocked: [],
+            columns: [
+                { text: "Tool Name", dataIndex: "tool_name" },
+                { text: "Tool ID", dataIndex: "tool_id" },
+                { text: "Action", dataIndex: "action" },
+            ],
+            toolshedColumns: [
+                { text: "Tool Name", dataIndex: "tool_name" },
+                { text: "Toolshed" },
+                { text: "Owner" },
+                { text: "Repository" },
+                { text: "Tool" },
+                { text: "Version" },
+            ],
+            message: "",
+            status: "",
+        };
+    },
+
+    components: {
+        message: Message,
+        "base-grid": BaseGrid,
+    },
+
+    created() {
+        axios
+            .get(`${getAppRoot()}api/sanitize_allow`)
+            .then((response) => {
+                this.isLoaded = true;
+                this.localAllowed = response.data.data.allowed_local;
+                this.localBlocked = response.data.data.blocked_local;
+                this.toolshedAllowed = response.data.data.allowed_toolshed;
+                this.toolshedBlocked = response.data.data.blocked_toolshed;
+                this.message = response.data.message;
+                this.status = response.data.status;
+            })
+            .catch((error) => {
+                console.error(error);
+            });
+    },
+
+    methods: {
+        allowHTML(tool_id) {
```

Comment:
That seems to be more than just a tool_id, can we split this up into what thing this really is ?